x-pack/plugins/ml/public/application/model_management/delete_models_modal.tsx

```diff
@@ -80,6 +102,55 @@ export const DeleteModelsModal: FC<DeleteModelsModalProps> = ({ modelIds, onClos
         </EuiModalHeaderTitle>
       </EuiModalHeader>
 
+      {modelsWithPipelines.length > 0 ? (
+        <EuiModalBody>
+          <EuiCallOut
+            title={
+              <FormattedMessage
+                id="xpack.ml.trainedModels.modelsList.deleteModal.pipelinesWarningHeader"
+                defaultMessage="{modelsCount, plural, one {{modelId} has} other {# models have}} associated pipelines."
+                values={{
+                  modelsCount: modelsWithPipelines.length,
+                  modelId: modelsWithPipelines[0].model_id,
+                }}
+              />
+            }
+            color="warning"
+            iconType="warning"
+          >
+            <div>
+              <p>
+                <FormattedMessage
+                  id="xpack.ml.trainedModels.modelsList.deleteModal.warningMessage"
+                  defaultMessage="Deleting the trained model and its associated {pipelinesCount, plural, one {pipeline} other {pipelines}} will result in the permanent removal of these resources. Any process configured to send data to the {pipelinesCount, plural, one {pipeline} other {pipelines}} will no longer be able to do so once you delete the {pipelinesCount, plural, one {pipeline} other {pipelines}}."
+                  values={{ pipelinesCount }}
+                />
+              </p>
+              <EuiCheckbox
+                id={'delete-model-pipelines'}
+                label={
+                  <FormattedMessage
+                    id="xpack.ml.trainedModels.modelsList.deleteModal.approvePipelinesDeletionLabel"
+                    defaultMessage="Delete {pipelinesCount, plural, one {pipeline} other {pipelines}}?"
```

Comment:
Nit, but looking at the writing guide, the examples don't include the `?` at the end of the label for the checkbox:

<img width="218" alt="image" src="https://github.com/elastic/kibana/assets/7405507/c828ebd6-3bc7-4107-b6ea-ce48333818fa">

Should we drop the `?` here @szabosteve ?

Reply:
Yes, you are right, we should.

Reply:
removed in https://github.com/elastic/kibana/pull/158671/commits/404f98b8b324f74c12be317e0b98263543aa3cb0